### ISSUE TITLE
keras2onnx.conver_keras

### ISSUE THREAD

**Issue by natank1** [onnx/keras-onnx#508]
I am using model combine as follow:
model1=load_model
model2=load_model

model2.layers[-1]="dense_2"

shape0= (20,100,1)
cominp= Input(shape=shape0,name="start_inpt")
model2.pop(0)
model1.pop(0)
y1=model1(cominp)
y2=model1(cominp)
mss1= Model(inputs=[cominp], outputs=[ y1,y2], name="comb_model")
When I use keras it works .
When I perform 
 keras2onnx.convert_keras(mss1)

I got the error :
"Cannot find the node for the model layer model_1"

I know it is some how related to "dense_1_1" But every modfication I try to do , brings me another bug

**Comment by jiafatom** [comment]
Can you provide a simple model for us to debug? thanks.

**Comment by natank1** [comment]
Sure

The problem comes when two models are loaded during different runs. :

e.g.:in one run
import onnx

inp0= Input(shape=(100,))

x = Dense(50)(inp0)

x= Dense(20)(x)
x = Dense(1,activation="sigmoid")(x)
mm1=Model(inputs=[inp0],outputs=[x])
mm1.compile(loss='binary_crossentropy', optimizer="RMSprop",
metrics=['accuracy'])
mm1.save("hh1.h5")

In the next run

inp0= Input(shape=(100,))

x = Dense(50)(inp0)

x= Dense(20)(x)
x = Dense(1,activation="sigmoid")(x)
mm1=Model(inputs=[inp0],outputs=[x])
mm1.compile(loss='binary_crossentropy', optimizer="RMSprop",
metrics=['accuracy'])
mm1.save("hh3.h5")

Make the models in different runs!!

Now do the following:

mm1 =load_model("hh1.h5")
mm1.layers.pop(0)

mm2 =load_model("hh3.h5")
mm2.layers.pop(0)


inp2= Input(shape=(100,),name ="comb_inpu")
y0= mm1(inp2)
y1= mm2(inp2)

mcomb=  Model(inputs=[inp2], outputs=[y0,y1], name="comb_model")
mcomb.compile(loss='binary_crossentropy', optimizer="RMSprop",
metrics=['accuracy'])
onnx_model = keras2onnx.convert_keras(mcomb)


The bug occurs in the last call

Thanks

Natan

**Comment by jiafatom** [comment]
I use your script here in our UT `test_layers.py`
```python
def test_multiple(runner):
    from tensorflow.keras.models import load_model
    inp0 = Input(shape=(100,))

    x = Dense(50)(inp0)

    x = Dense(20)(x)
    x = Dense(1, activation="sigmoid")(x)
    mm1 = Model(inputs=[inp0], outputs=[x])
    mm1.compile(loss='binary_crossentropy', optimizer="RMSprop",
                metrics=['accuracy'])
    mm1.save("hh1.h5")

    inp0 = Input(shape=(100,))

    x = Dense(50)(inp0)

    x = Dense(20)(x)
    x = Dense(1, activation="sigmoid")(x)
    mm1 = Model(inputs=[inp0], outputs=[x])
    mm1.compile(loss='binary_crossentropy', optimizer="RMSprop",
                metrics=['accuracy'])
    mm1.save("hh3.h5")

    mm1 = load_model("hh1.h5")
    mm1.layers.pop(0)

    mm2 = load_model("hh3.h5")
    mm2.layers.pop(0)

    inp2 = Input(shape=(100,), name="comb_inpu")
    y0 = mm1(inp2)
    y1 = mm2(inp2)

    mcomb = Model(inputs=[inp2], outputs=[y0, y1], name="comb_model")
    mcomb.compile(loss='binary_crossentropy', optimizer="RMSprop",
                  metrics=['accuracy'])
    onnx_model = keras2onnx.convert_keras(mcomb)
```
actually it works for me. 

When you try, are you using a tf.keras model? which keras2onnx version you use? Please try 1.7.0, or the master branch.

**Comment by faxu** [comment]
is further assistance needed? please try the latest converter and close the issue if it's resolved.

**Comment by natank1** [comment]
iT has been resolved

thanks
Natan
